vcs/gitcmd/repo.go

```diff
@@ -401,6 +409,12 @@ func (r *Repository) commitLog(opt vcs.CommitsOptions) ([]*vcs.Commit, uint, err
 
 	// Count commits.
 	cmd = exec.Command("git", "rev-list", "--count", rng)
+	if opt.Path != "" {
+		// TODO: This doesn't include --follow flag because rev-list doesn't support it, so the number may be slightly off.
+		//       Consider removing the "total" return value from the API altogether, that's why I'm not trying to do more to produce an accurate number here.
+		//       Getting the total number of commits is an expensive operation and likely isn't needed, or deserves a separate dedicated API if needed.
```

Comment:
I propose we make the breaking API change sooner, in order to improve the API (IMO), and remove `total` from `Commits`.

It belongs in a separate API call IMO (which we can add later, as needed), since it is a lot of work and callers of `Commits` will very often not use the value. This means expensive calculations are often wasted.